bot/cogs/moderation/silence.py

```diff
@@ -76,6 +47,21 @@ async def _get_instance_vars(self) -> None:
         self._mod_log_channel = self.bot.get_channel(Channels.mod_log)
         self._get_instance_vars_event.set()
 
+    async def schedule_unsilence(self, channel: SilencedChannel) -> None:
+        """Schedule expiration for silenced channels."""
+        await self.bot.wait_until_guild_available()
+        log.debug("Scheduling unsilencer")
+        self.schedule_task(channel.id, channel)
```

Comment:
This seems ineffective; the guild could become unavailable between being scheduled and when the sleep in the task finishes. It's also redundant; the scheduled task invokes the unsilence command, which already waits for `_get_instance_vars_event`.

Removing this wait, `schedule_unsilence` becomes redundant since it just does a simple log (which is also redundant since the scheduler already logs this) and calls one function.